lib/twilioconnection.js

```diff
@@ -30,14 +30,16 @@ const states = {
 
 const DEFAULT_MAX_CONSECUTIVE_MISSED_HEARTBEATS = 3;
 const DEFAULT_MAX_REQUESTED_HEARTBEAT_TIMEOUT = 5000;
+const DEFAULT_OPEN_TIMEOUT = 15000;
 const DEFAULT_WELCOME_TIMEOUT = 5000;
-const HEARTBEAT_TIMEOUT_OFFSET = 100;
+const OUTGOING_HEARTBEAT_OFFSET = 200;
 const WS_CLOSE_NORMAL = 1000;
 const WS_CLOSE_WELCOME_TIMEOUT = 3000;
 const WS_CLOSE_HEARTBEATS_MISSED = 3001;
 const WS_CLOSE_HELLO_FAILED = 3002;
 const WS_CLOSE_SEND_FAILED = 3003;
 const WS_CLOSE_NETWORK_CHANGED = 3004;
+const WS_CLOSE_FAILED_TO_OPEN = 3005;
```

Comment:
@makarandp0 Nitpick:
```suggestion
const WS_CLOSE_OPEN_TIMEOUT = 3005;
```